### PR TITLE
Add certificates to swift-ring-rebalance job

### PR DESCRIPTION
**Pull request by cschwede**
Follow up patches require the CA bundle to be available in the swift-ring-rebalance job.

### CODE REVIEW

**Comment by stuggi** [comment]
if it only requires the CA, it would be enough to add the CA as e.g. in https://github.com/openstack-k8s-operators/cinder-operator/blob/main/api/v1beta1/cinderscheduler_types.go#L64  and then the top level swift controller can assign the CA based on the input to the swift proxy as in https://github.com/openstack-k8s-operators/cinder-operator/blob/main/controllers/cinder_controller.go#L1073 . with this there would be no need to pass in the CA from the openstack-operator. does that make sense?

File: pkg/swiftring/job.go [hunk 1]
```diff
@@ -42,6 +42,15 @@ func GetRingJob(instance *swiftv1beta1.SwiftRing, labels map[string]string) *bat
 	envVars["OWNER_UID"] = env.SetValue(string(instance.ObjectMeta.UID))
 	envVars["OWNER_NAME"] = env.SetValue(instance.ObjectMeta.Name)
 
+	volumes := getRingVolumes(instance)
+	volumeMounts := getRingVolumeMounts()
+
+	// add CA cert if defined
+	if instance.Spec.TLS.CaBundleSecretName != "" {
```
Comment (stuggi) [inline pkg/swiftring/job.go:49]: swiftring controller should have an rbac for secrets

Comment (cschwede) [reply pkg/swiftring/job.go:49]: Right, adding the RBAC rule

**Comment by stuggi** [comment]
lgtm, should we add an assert of the swiftring and its job in the tls kuttl test? https://github.com/openstack-k8s-operators/swift-operator/blob/main/tests/kuttl/tests/basic-deploy_tls/01-assert-deploy-swift.yaml

**Comment by stuggi** [comment]
could also do this in a follow up if you prefer

**Comment by cschwede** [comment]
> lgtm, should we add an assert of the swiftring and its job in the tls kuttl test? https://github.com/openstack-k8s-operators/swift-operator/blob/main/tests/kuttl/tests/basic-deploy_tls/01-assert-deploy-swift.yaml

Yes, will do that in a follow up - going to add quite a bit of additional kuttl tests next

**Review by stuggi — APPROVED** [review]
/lgtm

**Pull request merged**